pkg/utils/etcdutil/health_checker.go

```diff
@@ -286,18 +286,21 @@ func (checker *healthChecker) updateEvictedEps(lastEps, pickedEps []string) {
 		}
 		return true
 	})
-	// Find all endpoints which are in the lastEps but not in the pickedEps,
-	// and add them to the evictedEps.
+	// Find all endpoints which are in `lastEps` and `healthyClients` but not in `pickedEps`,
+	// and add them to the `evictedEps`.
 	for _, ep := range lastEps {
 		if pickedSet[ep] {
 			continue
 		}
+		if hc := checker.loadClient(ep); hc == nil {
```

Comment:
This PR seems to be trying to solve the issue of non-atomic updates between `evictedEps` and the `client`. However, this PR doesn't fully address the problem.

For example: Do we need to check `if hc := checker.loadClient(ep); hc ==` for other `checker.evictedEps.Store(ep, 0)`?

Reply:
If the passed-in `lastEps` can be modified, is the returned `lastEps` also possibly inaccurate?

https://github.com/tikv/pd/blob/13174b5d4cab4d958f0b4ea2718ea7bb74992bc7/pkg/utils/etcdutil/health_checker.go#L210-L218